x-pack/plugins/canvas/public/components/arg_form/arg_template_form.tsx

```diff
@@ -5,17 +5,20 @@
  * 2.0.
  */
 
-import React, { useState, useEffect, useCallback, useRef } from 'react';
+import React, { useState, useEffect, useCallback, useRef, RefObject } from 'react';
 import usePrevious from 'react-use/lib/usePrevious';
+import useEffectOnce from 'react-use/lib/useEffectOnce';
 import { RenderToDom } from '../render_to_dom';
 import { ExpressionFormHandlers } from '../../../common/lib/expression_form_handlers';
+import { UpdatePropsRef } from '../../../types/arguments';
 
 interface ArgTemplateFormProps {
   template?: (
     domNode: HTMLElement,
     config: ArgTemplateFormProps['argumentProps'],
-    handlers: ArgTemplateFormProps['handlers']
-  ) => void;
+    handlers: ArgTemplateFormProps['handlers'],
+    onDone?: (ref: UpdatePropsRef) => void
```

Comment:
Is this onDone used anywhere?

Reply:
I agree, that I should change the name of this function from `onDone` to `onMount`) 

That is the key idea of the change I've done to support updating of the `argument` component props.

As you can see, at this line it is used for assigning the ref of mounted component with `updateProps` method to  the ref at the component, which is mounting template to some node. 
https://github.com/elastic/kibana/blob/eb695cfc0d7304f7b49f1db9cb1290246b76a7b4/x-pack/plugins/canvas/public/components/arg_form/arg_template_form.tsx#L62

And after this ref is used to update props of the component, rendered by the template.
https://github.com/elastic/kibana/blob/eb695cfc0d7304f7b49f1db9cb1290246b76a7b4/x-pack/plugins/canvas/public/components/arg_form/arg_template_form.tsx#L94

And here is a usage of the `onDone` method inside template: https://github.com/elastic/kibana/blob/eb695cfc0d7304f7b49f1db9cb1290246b76a7b4/x-pack/plugins/canvas/public/lib/template_from_react_component.tsx#L70

Reply:
I'll revisit this part of the functionality and change it to support a single form.